src/Stratis.Bitcoin/FinalizedBlockInfoRepository.cs

```diff
@@ -57,20 +59,23 @@ public class FinalizedBlockInfoRepository : IFinalizedBlockInfoRepository
 
         private readonly AsyncManualResetEvent queueUpdatedEvent;
 
-        public FinalizedBlockInfoRepository(IKeyValueRepository keyValueRepo, ILoggerFactory loggerFactory)
+        public FinalizedBlockInfoRepository(IKeyValueRepository keyValueRepo, ILoggerFactory loggerFactory, IAsyncProvider asyncProvider)
         {
             Guard.NotNull(keyValueRepo, nameof(keyValueRepo));
             Guard.NotNull(loggerFactory, nameof(loggerFactory));
 
             this.logger = loggerFactory.CreateLogger(this.GetType().FullName);
 
             this.keyValueRepo = keyValueRepo;
+            this.asyncProvider = asyncProvider;
```

Comment:
no need to save async provider, it can be a local variable.

Reply:
force of habit

Reply:
Though this make me realize we should probably not start Tasks in class constructors (that's why we use a `Start` method normally)